enhancements/monitoring/scrape-profiles.md

```diff
@@ -0,0 +1,274 @@
+  // Defines the scraping profile that will be enforced on the platform
+  // prometheus instance. Possible values are full and minimal.
+  //
+  // +kubebuilder:validation:Enum=full;minimal
+  // +optional
+	ScrapeProfile string `json:"scrapeProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+Each OpenShift team that wants to adopt this feature will be responsible for providing the different monitors. This work is not trivial. Dependencies between operators and their metrics exist. This makes it difficult for developers to determine whether a given metric must be added to the "minimal" profile or not. To aid teams with this problem we will provide a tool that would consume a monitor and generate the "minimal" monitor for that operator. However for this, the tool would need to have access to an up to date, installation of OpenShift in order to query the Prometheus instance running in the cluster.
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't being used in an alert is now required, how does the monitor responsible for that metric gets updated?
+  - The tool we provide would run in CI and ensure are ServiceMonitors are up to date;
+
+- A new profile is added, what will happen to operators that had implemented scrape profiles but did not implement the latest profile.
+  - TBD. Currently, according to the description above, they would not be scraped when the new profile would be picked.
+
+### Drawbacks
+
+- Extra CI cycles
+
+## Design Details
+
+### Open Questions [optional]
+
+- Should we add future profiles? 
+- Should developers have to comply with all profiles?
+
+### Test Plan
+
+- Unit tests in CMO to validate that the correct monitors are being selected
+- E2E tests in CMO to validate that everything works correctly
+Unsure on this one... (- Testing in openshift/CI to validate that every metrics being used exists in the cluster?)
+
+### Graduation Criteria
+
+Plan to release as Dev Preview initially.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+- Upgrade and downgrade are not expected to present a significant challenge. The new field is a small layers over existing stable APIs.
```

Comment:
At least, we need to think about the 4.12 > 4.13 upgrade path. Some of the operators may be upgraded before CMO and there may be a short period of time when the Prometheus operator would reconcile all monitoring objects irrespective of their scrape profile.